server/client.go

```diff
@@ -652,7 +666,12 @@ func (c *client) setPermissions(perms *Permissions) {
 			c.perms.sub.allow = NewSublistWithCache()
 		}
 		for _, subSubject := range perms.Subscribe.Allow {
-			sub := &subscription{subject: []byte(subSubject)}
+			s, q := splitSubject(subSubject)
```

Comment:
Maybe have this helper function return []byte, []byte since we always cast below?